core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ConcurrencyController.java

```diff
@@ -38,8 +35,6 @@ public void update(double utilization, int currentConcurrency)
         else if (utilization < TARGET_UTILIZATION && currentConcurrency >= targetConcurrency) {
             targetConcurrency++;
         }
-
-        targetConcurrency = Math.min(maxConcurrency, targetConcurrency);
```

Comment:
does this undo https://github.com/trinodb/trino/pull/20666?

Reply:
The implementation - yes. The logic is preserved, but it's now done somewhere else.